docs/source/multi_gpu_train.md

```diff
@@ -0,0 +1,54 @@
+## Multi-GPU data-parallel training using the `Trainer` class
+
+To train models faster, users can use Data-Parallel training when using `transformers4rec.Trainer` class for training. When using the data-parallel approach for multi-GPU training, the training data will be distributed between workers to speedup training and support larger batch sizes within each step of training.
+
+The `Trainer` class supports both `DataParallel` and `DistributedDataParallel` built-in features of PyTorch. in this documentation we will explain how each of these built-in features work and how they can be used.
+
+### DataParallel
+When the DataParallel mode is used for training on each training step:
+- GPU0 reads the full batch then evenly distributes it between available GPUs
+- The latest model will be copied to all GPUs
+- A Python thread will be assigned to running `forward()` on each GPU and thepartial loss will be sent to GPU0 to compute total loss
+- Computed total loss will be scattered between GPUs and threads will run `backward()`
+- Gradients from each GPU will be sent to GPU0 and their average will be computed
+
+As we see, parallelism in `DataParallel` mode is implemented through Python threads which will be blocked by GIL (Global Interepreter Lock) so `DataParallel` is not the preferred method. Users are advised to use `DistributedDataParallel` instead as it uses multi-processing instead of multi-threading and is better maintained. Also, some models types such as `transfoxl` can not be used with `DataParallel`. To learn more about `DataParalle` refer to [PyTorch documentations](https://pytorch.org/docs/master/generated/torch.nn.DataParallel.html).
```

Comment:
```suggestion
As we see, parallelism in `DataParallel` mode is implemented through Python threads which will be blocked by GIL (Global Interepreter Lock) so `DataParallel` is not the preferred method. Users are advised to use `DistributedDataParallel` instead as it uses multi-processing instead of multi-threading and is better maintained. Also, some models types such as `transfoxl` can not be used with `DataParallel`. To learn more about `DataParallel` refer to [PyTorch documentation](https://pytorch.org/docs/master/generated/torch.nn.DataParallel.html).
```